File: hidrogine.lwjgl/src/hidrogine/lwjgl/Material.java
package hidrogine.lwjgl;



public class Material {
    public Float[] Ka;
    public Float[] Kd;
    public Float[] Ks;
    public Float Tf;
    public Float illum;
    public Float d;
    public Float Ns;
    public Float sharpness;
    public Float Ni;
    public int texture = 0;
    public String name;
    
    public Material(String n) {
        name = n;
    }



    
    
    public void setTexture(String filename) {
        texture = TextureLoader.loadTexture(filename);
    }





    public Object getName() {
        return name;
    }
}

File: Windrogine/hidrogine/hidrogine/hidrogine/Material.cpp
#include "Material.h"

namespace hidrogine {
    Material::Material(string n) {
        name = n;
    }
    
    void Material::setTexture(string filename) {
        texture = TextureLoader::loadTexture(filename);
    }

    string Material::getName() {
        return name;
    }
}